### ISSUE TITLE
Date not correctly decoded on PATCH/POST

### ISSUE THREAD

**Issue by nickschot** [postlight/lux#621]
Date not correctly decoded for at least PATCH and POST requests. Date is sent as an ISO 8601 string in the JSON (exactly the way it was received from Lux) and defined as a date type in the migration.

Example: `"2013-02-07T00:00:00.000Z"` is decoded as `{}` in the controller.

Relevant stacktrace:
```sh
TypeError: Expected type 'date' for parameter 'data.attributes.date' but got 'object'.
    at validateType$1 (/node_modules/lux-framework/src/packages/router/route/params/utils/validate-type.js:46:5)
    at Set.validate (/node_modules/lux-framework/src/packages/router/route/params/parameter/index.js:40:5)
    at Map.validate (/node_modules/lux-framework/src/packages/router/route/params/parameter-group/index.js:59:45)
    at Map.validate (/node_modules/lux-framework/src/packages/router/route/params/parameter-group/index.js:59:45)
    at Map.validate (/node_modules/lux-framework/src/packages/router/route/params/parameter-group/index.js:59:45)
    at /node_modules/lux-framework/src/packages/router/route/index.js:157:28
    at Generator.next (<anonymous>)
    at step (/dist/bundle.js:76:30)
    at /dist/bundle.js:94:14
    at /dist/bundle.js:73:12
    at Set.visit (/node_modules/lux-framework/src/packages/router/route/index.js:149:58)
    at parseRequest.then.params (/node_modules/lux-framework/src/packages/server/index.js:119:26)
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

**Comment by nickschot** [comment]
This is a nasty one. The date is parsed correctly all the way to the end of the `format` function in `/src/packages/server/request/parser/utils/format.js`. There `camelizeKeys` (part of the 'inflection' npm package) is called which transforms: 

```javascript
{
 'birth-date': 2017-01-15T00:00:00.000Z 
}
```
into 
```javascript
{ 
    birthDate: {} 
}
```

"2017-01-15T00:00:00.000Z" in the first object is a valid `Date` object.

Separately from all the above it is (I do have to double check this) also necessary to add an extra case to `validateType` in `/src/packages/router/route/params/utils/validate-type.js` due to `typeof new Date() === 'object'`.

This makes it:
```javascript
switch (type) {
    case 'array':
      isValid = Array.isArray(value);
      break;

    case 'buffer':
      isValid = isBuffer(value);
      break;

    case 'object':
      isValid = isObject(value) || isNull(value);
      break;

    case 'date':
      isValid = value instanceof Date;
      break;

    default:
      isValid = type === valueType;
  }
```

**Comment by zacharygolba** [comment]
Closed by #639
